pkg/dockerregistry/server/repository.go

```diff
@@ -133,18 +132,16 @@ func (r *repository) Manifests(ctx context.Context, options ...distribution.Mani
 	ms = &manifestService{
 		repo:          r,
 		manifests:     ms,
-		acceptschema2: r.config.acceptSchema2,
+		acceptschema2: r.app.extraConfig.Compatibility.AcceptSchema2,
 	}
 
-	if r.config.pullthrough {
+	if r.app.extraConfig.Pullthrough.Enabled {
```

Comment:
Please use `r.app.extraConfig` instead of `r.enabledMetrics` as well.

Reply:
done